neuro_data/static_images/data_schemas.py

```diff
@@ -342,15 +342,52 @@ def load_frame(self, key):
         elif stimulus.TrippyFrame & key:
             assert (stimulus.TrippyFrame & key).fetch1('pre_blank_period') > 0, 'we assume blank periods'
             return (stimulus.TrippyFrame & key).fetch1('img')
-        if stimulus.ColorFrameProjector & key:
+        elif stimulus.ColorFrameProjector & key:
+            # stimulus is type ColorFrameProjector which means we need to look up what channel was map to what and select base on
             assert (stimulus.ColorFrameProjector & key).fetch1('pre_blank_period') > 0, 'we assume blank periods'
-            return (stimulus.StaticImage.Image & (stimulus.ColorFrameProjector & key)).fetch1('image')
+
+            original_img = (stimulus.StaticImage.Image & (stimulus.ColorFrameProjector & key)).fetch1('image')
+            if len(original_img.shape) == 2:
+                # Only 1 channel
+                return original_img
+            else:
+                # There is more then 1 channel, thus we need get the channel mappings for the project, where the number signifies which RGB channel maps to the project channels
+                channel_mappings = (stimulus.ColorFrameProjector() & key).fetch1('channel_1', 'channel_2', 'channel_3')
+                image_sub_channels_to_include = []
+                for channel_mapping in channel_mappings:
+                    if channel_mapping is not None:
+                        image_sub_channels_to_include.append(original_img[:, :, channel_mapping - 1])
+                return np.stack(image_sub_channels_to_include).transpose(1, 2, 0)
```

Comment:
I think you can call npo.stack([f1, f2, f3], axis=-1) and it will stack it in the last axis, to avoid the transpose

Reply:
That works better, less function calls. Thanks